zerver/lib/actions.py

```diff
@@ -5570,6 +5584,14 @@ def get_service_dicts_for_bots(bot_dicts: List[Dict[str, Any]],
                 service_dicts = [{'config_data': bot_config,
                                   'service_name': services[0].name
                                   }]
+        elif bot_type == UserProfile.INCOMING_WEBHOOK_BOT:
+            # Even though Incoming Webhook Bots don't have any "services" attached, we use
+            # "service_dicts" to keep things standard and compatible with outgoing webhook
+            # bots and embedded bots. Our only aim for incoming webhook bots is to convey
+            # the bot's configuration data.
+            if bot_profile_id in incoming_webhook_bot_configs.keys():
+                bot_config = incoming_webhook_bot_configs[bot_profile_id]
+                service_dicts = [{'config_data': bot_config}]
```

Comment:
Hmm, could we avoid duplicating the `embedded_bot_ids` code path and instead do `elif bot_type in [UserProfile.INCOMING_WEBHOOK_BOT, Profile.EMBEDDED_BO]` both here and above?  It seems like the only difference here is the `service_name` block.  And we could save an unnecessary database query.

Reply:
I modified my code to remove the extra database query (i.e. the extra call to `get_bot_configs`).